File: StudentCredit.Application/Credits/Dtos/CreditInfoDto.cs
using StudentCredit.Application.Applications.ApplicationsTwo.Dtos;
using StudentCredit.Data.Applications.ApplicationOne.Enums;
using StudentCredit.Data.Applications.Common.Enums;
using StudentCredit.Data.Common.Enums;
using StudentCredit.Infrastructure.Helpers.Dtos;

namespace StudentCredit.Application.Credits.Dtos
{
    public class CreditInfoDto
	{
		public int Id { get; set; }
		public CommitState CommitState { get; set; }
		public bool? PaidByApplicationFour { get; set; }
		public NomenclatureDto ApplicationOneType { get; set; }

		public string StudentFullName { get; set; }
		public string Uin { get; set; }
		public NomenclatureDto Institution { get; set; }
		public NomenclatureDto ResearchArea { get; set; }
		public NomenclatureDto Speciality { get; set; }
		public NomenclatureDto EducationalQualification { get; set; }
		public string Uan { get; set; }
		public EducationType EducationType { get; set; }
		public DateTime? BirthDate { get; set; }
		public ExistingEnum? SpecialityEnum { get; set; }
		public string MigrationSpecialityName { get; set; }

		public ExistingEnum? ResearchAreaEnum { get; set; }
		public string MigrationResearchAreaName { get; set; }

		public NomenclatureDto Bank { get; set; }

		public DateTime? ContractDate { get; set; }
		public string CreditNumber { get; set; }
		public CreditType CreditType { get; set; }
		public DateTime? ExpirationDateOfGracePeriod { get; set; }
		public int? PaymentPeriod { get; set; }
		public decimal? CreditSize { get; set; }

		public List<CreditApplicationDto> ApplicationsOne { get; set; } = new List<CreditApplicationDto>();
		public List<RecordEntryDto> ApplicationsTwo { get; set; } = new List<RecordEntryDto>();
		public List<CreditApplicationDto> ApplicationsFour { get; set; } = new List<CreditApplicationDto>();
	}
}

File: WebApplication/src/modules/application/credit/models/credit-info.dto.ts
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { EducationType } from "../../application-one/enums/education-type.enum";
import { CreditType } from "../../common/enums/credit-type.enum";
import { CreditApplicationDto } from "./credit-application.dto";
import { RecordEntryDto } from "src/modules/application-two/models/record-entry.dto";
import { CommitState } from "../../common/enums/commit-state.enum";
import { ExistingEnum } from "../../common/enums/existing.enum";

export class CreditInfo {
  id: number;
  commitState: CommitState;
  paidByApplicationFour: boolean;

  studentFullName: string;
  uin: string;
  institution: NomenclatureDto;
  researchArea: NomenclatureDto;
  speciality: NomenclatureDto;
  educationalQualification: NomenclatureDto;
  uan: string;
  educationType: EducationType;
  birthDate: Date;
  specialityEnum: ExistingEnum;
  migrationSpecialityName: string;
  researchAreaEnum: ExistingEnum;
  migrationResearchAreaName: string;

  bank: NomenclatureDto;

  contractDate: Date | null;
  creditNumber: string;
  creditType: CreditType;
  expirationDateOfGracePeriod: Date | null;
  paymentPeriod: number;
  creditSize: number | null;

  applicationsOne: CreditApplicationDto[] = [];
  applicationsTwo: RecordEntryDto[] = [];
  applicationsFour: CreditApplicationDto[] = [];
}